bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/console/BoschShcCommandExtension.java

```diff
@@ -0,0 +1,275 @@
+        super(BoschSHCBindingConstants.BINDING_ID, "Interact with the Bosch SmartHomeController.");
+        this.thingRegistry = thingRegistry;
+    }
+
+    private List<String> findAllBoschShcServices() {
+
+        // Getting all services via class loader doesn't work for me,
+        // because some service classes like airqualitylevel are not listed until used.
+
+        // return Arrays.stream(this.getClass().getClassLoader().getDefinedPackages())
+        // .filter(p -> p.getName().startsWith("org.openhab.binding.boschshc.internal.services."))
+        // .filter(p -> !p.getName().endsWith("dto"))
+        // .map(p -> p.getName().replace("org.openhab.binding.boschshc.internal.services.", "")).sorted()
+        // .filter(s -> !s.isEmpty()).collect(Collectors.toList());
+
+        return List.of("airqualitylevel", "batterylevel", "binaryswitch", "cameranotification", "childlock",
+                "hsbcoloractuator", "humiditylevel", "intrusion", "latestmotion", "multilevelswitch", "powermeter",
+                "powerswitch", "privacymode", "roomclimatecontrol", "shuttercontact", "shuttercontrol", "silentmode",
+                "smokedetectorcheck", "temperaturelevel", "valvetappet");
+    }
+
+    @Override
+    public void execute(String[] args, Console console) {
+        if (args.length == 0) {
+            printUsage(console);
+            return;
+        }
+
+        try {
+
+            if (GET_BRIDGEINFO.equals(args[0])) {
+                console.print(printBridgeInfo().toString());
+                return;
+            }
+
+            if (GET_DEVICES.equals(args[0])) {
+                console.print(printDeviceInfo().toString());
+                return;
+            }
+
+            if (SHOW_BINDINGINFO.equals(args[0])) {
+                console.print(printBindingInfo().toString());
+                return;
+            }
+            if (SHOW_DEVICES.equals(args[0])) {
+                console.print(printSupportedDevices().toString());
+                return;
+            }
+            if (SHOW_SERVICES.equals(args[0])) {
+                console.print(printSupportedServices().toString());
+                return;
+            }
+
+        } catch (BoschSHCException | ExecutionException | TimeoutException e) {
+            console.print(String.format("Error %1s%n", e.getMessage()));
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+
+        // unsupported command, print usage
+        printUsage(console);
+    }
+
+    private List<BridgeHandler> getBridgeHandlers() {
+        List<BridgeHandler> bridges = new ArrayList<>();
+        for (Thing thing : thingRegistry.getAll()) {
+            ThingHandler thingHandler = thing.getHandler();
+            if (thingHandler instanceof BridgeHandler bridgeHandler) {
+                bridges.add(bridgeHandler);
+            }
+        }
+        return bridges;
+    }
+
+    StringBuilder printBridgeInfo()
```

Comment:
The method names `print*()` suggest that there is a side effect that prints to the console, but in fact the methods do not perform the printing, the merely return a string(builder). I would suggest to rename all `print*()` methods to something like `build*(), `create*()`, `construct*()`. Also: is there any advantage to return the `StringBuilder` instead of returning the `String` (`builder.toString()`) directly?